pkg/models/registry.go

```diff
@@ -96,14 +108,28 @@ func (r *LocalRegistry) MessageTypeRegistry() *protoregistry.Types {
 	for _, model := range r.modelNames {
 		typeRegistry.RegisterMessage(dynamicpb.NewMessageType(model.proto.ProtoReflect().Descriptor()))
 	}
+	proxiedTypes := r.proxied.MessageTypeRegistry()
+	proxiedTypes.RangeMessages(func (mt protoreflect.MessageType) bool {
+		typeRegistry.RegisterMessage(mt)
+		return true
+	})
 	return typeRegistry
 }
 
 // Register registers a protobuf message with given model specification.
 // If spec.Class is unset empty it defaults to 'config'.
 func (r *LocalRegistry) Register(x interface{}, spec Spec, opts ...ModelOption) (KnownModel, error) {
-	goType := reflect.TypeOf(x)
+	// check if the model was learned remotely
+	if modelInfo, isProxied := x.(*ModelInfo); isProxied {
```

Comment:
The syntax of the "x" paramater doesn't inforce it, but the documentation says that the protobuf message is expected. So i would prefer a proto message as input. In your case you can use this to create new empty proto message:
`dynamicpb.NewMessageType(modelinfo.MessageDescriptor).New().Interface()`
(maybe you can put it somewhere as conversion helper function)
This will create a dynamicpb.Message instance and you can use that type here as type check switch.

Reply:
I can update the function documentation, but I don't think that I can use dynamicpb here since RemoteRegistry requires ModelInfo.

Reply:
Unfortunately true. So please just update to docs (the registry interface docs probably also need some update).

Reply:
done